gcm_filters/filter.py

```diff
@@ -252,10 +263,15 @@ def filter_func_vec(ufield, vfield, *args):
                     (utemp_b, vtemp_b) = laplacian(
                         utemp_l, vtemp_l
                     )  # Compute Biharmonic (apply Laplacian twice)
-                    ufield_bar += (
-                        utemp_l * 2 * np.real(s_b) / np.abs(s_b) ** 2
-                        + utemp_b * 1 / np.abs(s_b) ** 2
-                    )
+                    if Laplacian.is_nondimensional:
+                        utemp_l /= filter_spec.dx_min_sq  # dimensionalize
+                        vtemp_l /= filter_spec.dx_min_sq  # dimensionalize
+                        utemp_b /= filter_spec.dx_min_sq ** 2  # dimensionalize
+                        vtemp_b /= filter_spec.dx_min_sq ** 2  # dimensionalize
+                        ufield_bar += (
+                            utemp_l * 2 * np.real(s_b) / np.abs(s_b) ** 2
+                            + utemp_b * 1 / np.abs(s_b) ** 2
+                        )
```

Comment:
Lines 271-274  should not be part of the if-statement.

Reply:
Good catch! That would have been bad.

Reply:
Wondering why the tests did not catch this? :confused: